cpp/BUILD.bazel

```diff
@@ -14,11 +16,17 @@ cc_binary(
     target_compatible_with = [
         "@platforms//os:linux",
     ],
+    deps = ["@libsystemd"],
+)
+
+release_strip_binary(
```

Comment:
is this necessary to make the new toolchains work?

Reply:
There was a sandbox path leaking into the binary if we did not strip here.

I believe zig cc has a different default strip behavior. Before, we would always strip debug symbols (I think this is where the path was), and now we have to opt in.

Reply:
> There was a sandbox path leaking into the binary if we did not strip here.

Then this will come back to bite us later and will cause rebuilds. Let me know if you want me to have a look!

Reply:
Oh I think I see, do you mean that because the earlier `infogetty_bin` target isn't deterministic, it can cause us build issues?

We should be able to push the debug stripping into the same target, zig should support `-S`/`--strip-debug`: https://github.com/ziglang/zig/blob/24db007cde02f756c8825d0bab0ed46c104f5430/src/main.zig#L2569-L2570

Or do you think we should handle this even further down, so that the extra flags aren't required?

Reply:
> Oh I think I see, do you mean that because the earlier infogetty_bin target isn't deterministic, it can cause us build issues?

Yep, more or less!

> Or do you think we should handle this even further down, so that the extra flags aren't required?

Up to you, you have much more context here!

Reply:
How does this look?